Astro CLI 1.24 release notes and reference documentation (#3423)
astronomer/docs: astro/cli/release-notes.md

This document provides a summary of all changes made to the [Astro CLI](cli/overview.md). For general product release notes, go to [Astro Release Notes](release-notes.md). If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/open-support-request).

- **Stable versions**: 1.24.0, 1.23.0, and 1.22.0. See [Astro CLI release and lifecycle policy](cli/release-lifecycle-policy.md).

> ## Astro CLI 1.24.0

Author: Any other small changes or bug fixes we should include?

Reviewer: There a few more changes in this release:

- workload-identity flag and field for astro deployment create/update and the deployment file - https://github.com/astronomer/astro-cli/pull/1550
- new `astro organization role list` command - https://github.com/astronomer/astro-cli/pull/1554
- fixed an issue were `deployment variable list --save` did not work for secret variables - https://github.com/astronomer/astro-cli/pull/1558
- fixed an issue where updating a Deployment with a deployment file did not work with a deployment API token - https://github.com/astronomer/astro-cli/pull/1559
- new commands for user/team role management. Still trying to figure out if this will be in the release. It is main but I don't know much about it - https://github.com/astronomer/astro-cli/pull/1553

> Release date: February 19, 2024

Reviewer 2: ```suggestion
Release date: February 21, 2024
```
Did this change or are we publishing retroactively?

Reviewer: The release date will be Feburary 22nd

### Support for hibernating development Deployments

> You can now use the Astro CLI to hibernate or wake up a development Deployment. These commands work well in automated processes where a Deployment requires flexibility for when it hibernates. Note that you can hibernate a Deployment only if you enabled **Development Mode** when you [created the Deployment](create-deployment.md).

Reviewer 2: ```suggestion
You can now use the Astro CLI to hibernate or wake up a development Deployment. These commands work well in automated processes where a Deployment requires flexibility for when it hibernates. Note that you can hibernate a Deployment only if you enable **Development Mode** when you [create the Deployment](create-deployment.md).
```

Use the following new commands to hibernate development Deployments regardless of their existing hibernation schedule:

- [`astro deployment hibernate`](https://docs.astronomer.io/astro/cli/astro-deployment-hibernate)
- [`astro deployment wake-up`](https://docs.astronomer.io/astro/cli/astro-deployment-wake-up)

> You can additionally now create new development Deployments and configure long-term hibernation schedules for them using `astro deployment create`.

Author: ```suggestion
Additionally, you can create new development Deployments and configure long-term hibernation schedules for them using `astro deployment create`.
```

This document provides a summary of all changes made to the [Astro CLI](cli/overview.md). For general product release notes, go to [Astro Release Notes](release-notes.md). If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/open-support-request).

- **Stable versions**: 1.24.0, 1.23.0, and 1.22.0. See [Astro CLI release and lifecycle policy](cli/release-lifecycle-policy.md).

## Astro CLI 1.24.0

> Release date: February 22, 2024

Reviewer 2: ```suggestion
Release date: February 27, 2024
```
